recommender/static/js/src/recommender.js

```diff
@@ -1,852 +1,930 @@
+     */
+    function bindInterruptSubmissionEvent() {
+        var divs = $('.flagResourcePage, .editResourcePage, .addResourcePage, .deendorsePage, .endorsePage, .importResourcePage', element);
+        var activePage;
+        var logStudentInput = {};
+
+        for (var key in divs) {
+            if ($(divs[key]).is(':visible')) {
+                activePage = divs[key];
+                break;
             }
         }
-        var displayedDiv = findDisplayedBlock();
         
-        logDict = {'status': 'Back to resource list mode', 'element': $(element).attr('data-usage-id')};
-        function getTypedContent(selector) {
-            if ($(displayedDiv).find(selector).length != 0) {
-                $(displayedDiv).find(selector).each(function() {
-                    logDict[$(this).attr('class').replace('tooltipstered', '').trim()] = $(this).val();
-                });
+        logStudentInput = logTypedInformation('textarea', activePage, logStudentInput);
+        logStudentInput = logTypedInformation('input[type="text"]', activePage, logStudentInput);
+        logStudentInput = logTypedInformation('input[type="file"]', activePage, logStudentInput);
+        Logger.log('mit.recommender.backToView', generateLog(loggerStatus['backToView']['backToView'], logStudentInput));
+
+        var canGoBackToView = true;
+        if ($('input[type="button"]:disabled', activePage).length == 0) {
+            canGoBackToView = confirm(confirmInterruptSubmission)
+        }
+        if (canGoBackToView) { backToView(); }
+    }
+
+    /**
+     * Clear the previously given information in the page for adding resources.
+     */
+    function resetAddResourcePage() {
+        $('.addResourcePage', element).find('input[type="text"]').val('');
+        $('.addResourcePage', element).find('textarea').val('')
+        $('.addResourceForm', element).find("input[name='file']").val('');
+        $('.addSubmit', element).attr('disabled', true);
+    }
+
+    /**
+     * Check whether enough information (title/url) is provided for
+     * recommending a resource, if yes, enable the summission button.
+     */
+    function enableAddSubmit() {
+        if ($('.inTitle', element).val() == '' || $('.inUrl', element).val() == '') {
+            $('.addSubmit', element).attr('disabled', true);
+            return;
+        }
+        $('.addSubmit', element).attr('disabled', false);
+    }
+
+    /**
+     * Show a resource in the resource list.
+     * @param {number} votes The votes which the shown resource have.
+     * @param {dictionary} resource The resource to be shown.
+     * @returns {element} The element of shown resource.
+     */
+    function showResourceEntry(votes, resource) {
+        /* Decide the position for the added resource (pos), by sorting the votes */
+        var pos = -1;
+        $('.recommenderVoteScore', element).each(function(idx, ele){ 
+            if (parseInt($(ele).text()) < votes) {
```

Comment:
Always, always pass 10 as the second argument to `parseInt`. In some cases, it assumes base 8 by default.